### ISSUE TITLE
flac files register with <5 secs length, don't play [SOLVED]

### ISSUE THREAD

**Issue by thewaywest** [cmus/cmus#571]
FLAC issue -- research open and closed tickets, without finding a lead on this.

Newly installed cmus (from repo), reads music directory and finds but fails to correctly register FLAC. They are all listed with play times of <5 secs, and simply skip when encountered in the playlist. My favorites ripped at high quality, and I can't play them :(

I've read of an option to drop the flac plugin in favor of ffmpeg -- could be of help here, as I assume the plugin is used to grep meta out of the files as well as play them (true?).

Not sure where to find the ffmpeg plugin to test this. In any case, would prefer (as in other threads) to see the native flac support corrected, than apply bandaids.

Let me know if I can provide other information.

Thanks

1) `uname -a`:
```
Linux [  ] 4.4.0-45-generic #66-Ubuntu SMP Wed Oct 19 14:12:37 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
```

2) `cmus --version`:
```
cmus v2.7.1
Copyright 2004-2006 Timo Hirvonen
Copyright 2008-2013 Various Authors
```

3) `cmus --plugins`:
```
Input Plugins: /usr/lib/cmus/ip
  mad:
    Priority: 55
    File Types: mp3 mp2
    MIME Types: audio/mpeg audio/x-mp3 audio/x-mpeg
  opus:
    Priority: 50
    File Types: opus
    MIME Types:
  cue:
    Priority: 50
    File Types:
    MIME Types: application/x-cue
  vorbis:
    Priority: 50
    File Types: ogg oga ogx
    MIME Types: application/ogg audio/x-ogg
  wav:
    Priority: 50
    File Types: wav
    MIME Types:
  flac:
    Priority: 50
    File Types: flac fla
    MIME Types:
  cdio:
    Priority: 50
    File Types:
    MIME Types: x-content/audio-cdda
  wavpack:
    Priority: 50
    File Types: wv
    MIME Types: audio/x-wavpack
  modplug:
    Priority: 50
    File Types: mod s3m xm it 669 amf ams dbm dmf dsm far mdl med mtm okt ptm stm ult umx mt2 psm
    MIME Types:
  aac:
    Priority: 50
    File Types: aac
    MIME Types: audio/aac audio/aacp
  mpc:
    Priority: 50
    File Types: mpc mpp mp+
    MIME Types: audio/x-musepack
  ffmpeg:
    Priority: 30
    File Types: ac3 aif aifc aiff ape au mka shn tta wma aac fla flac m4a m4b mp+ mp2 mp3 mp4 mpc mpp ogg wav wv
    MIME Types:

Output Plugins: /usr/lib/cmus/op
  pulse
  alsa
  jack
  ao
```

**Comment by mahkoh** [comment]
Set input.priority.flac to 0 to disable the flac plugin. Duplicate of #440.

**Comment by thewaywest** [comment]
thanks. not duplicate, these are not small files

**Comment by thewaywest** [comment]
update --

$ mv /usr/lib/cmus/ip/flac.so /usr/lib/cmus/flac.so   #out of directory
cmus :update-cache -f
cmus :q and restart
---------------------------------
files still registering as under 5s each. do not play. VLC reads and plays without error.

**Comment by mahkoh** [comment]
You do not need to remove the plugin. Set the priority to 0, run update-cache -f, and then it should work.

**Comment by thewaywest** [comment]
thanks. new to cmus -- where to set param?

**Comment by mahkoh** [comment]
The settings view, normally bound to key 7.

**Comment by thewaywest** [comment]
yeah looked there --  input.cdio.cddb_url  in teh only option under "input", and no flac to be found. no command "priority". can you be more explicit? thanks for the help

**Comment by mahkoh** [comment]
Oh I'm sorry you've installed the last release which doesn't have this option yet. In this case you have to remove the so file manually, which you've already done.

In any case, after removing the so file restart cmus and then run update-cache -f.

**Comment by thewaywest** [comment]
yes, that's what I did above. rebooted, reran as stated, no change.

I am going to look at the tagging on those files. though files tags shouldn't be relied on for this sort of thing..?

**Comment by mahkoh** [comment]
Just to make sure:

1. You removed the .so file

2. Then you restarted cmus

3. Then you ran updated-cache -f

If this is the case please check that cmus --plugins no longer lists the flac plugin.

**Comment by thewaywest** [comment]
of course. flac.so is not there, all steps repeated (twice), but no change -- probably good to keep this bug open. and research further.

if there is more information i can provide from my install, happy to do so.

**Comment by mahkoh** [comment]
Can you send me one of the affected files?

**Comment by mahkoh** [comment]
If so please send it to the email address shown in this file: https://github.com/cmus/cmus/commit/37d2e24a7764bae2ccdc9957cd4640560ef43f03.patch

**Comment by thewaywest** [comment]
sure. I'll find or gen a small(er) one that proves not to work in current install. I might branch the code and lean in as well... given a little time... 

thanks for your work here.

**Comment by thewaywest** [comment]
sent

**Comment by mahkoh** [comment]
I've quickly tested the files you've sent me and they seem to play correctly with durations of 33s and 1m57s respectively. Is this the expected behavior?

I've tested this with both the last stable release (2.7.1) and the master branch. Both times I've used the flac plugin.

I'm not sure which Ubuntu version you're using but in the past we've had problems with the binaries distributed by Ubuntu. Maybe you can try to compile the current master branch. Don't forget that you've to `sudo make install` because otherwise cmus won't find the plugins.

**Comment by mahkoh** [comment]
I've taken another look at your email and you have one file which is rather small (175kb). So maybe this is related to #440. #440 will hopefully soon be fixed via #572.

We've also seen issues with duplicate files in the past when there were symlinks involved. Maybe you should try to add the files individually and see what happens then.

**Comment by thewaywest** [comment]
I've just found mp3 files where the browser (?) module is not finding all the files. my entire music directory is on another partition, mounted with fstab, and symlinked to my home directory. since Linux treats a symlinked -directory- as a local directory, it would be pretty sad if the browser couldn't track these files! (?)

sad for me, anyway, as I have multiple OS running on this device, and want to keep that partition separate for access and ease of backup....

I don't believe this is the small file issue. And I no longer believe it has anything to do with FLAC, though I am seeing the problem there much more frequently than elsewhere.

I installed from Xenial repo... I can (learn how to) build locally and do that, but better for distribution and adoption of this cool program would be if the repos were broken.

**Comment by mahkoh** [comment]
>I've just found mp3 files where the browser (?) module is not finding all the files. my entire music directory is on another partition, mounted with fstab, and symlinked to my home directory. since Linux treats a symlinked -directory- as a local directory, it would be pretty sad if the browser couldn't track these files! (?)

This is all very suspect. I've personally never had a problem with symlinks but I've seen other people report such issues.

>I installed from Xenial repo... I can (learn how to) build locally and do that, but better for distribution and adoption of this cool program would be if the repos were broken.

Unfortunately downstream is not very cooperative. For most people everything seems to work fine though. Building from source should be easy. It's explained in the readme. Since you're using ubuntu you'll have to separately install the development packages first.

**Comment by thewaywest** [comment]
built, but without all the dev pkg reqs (where's the list?) -- and even reinstalling from repo, something is whacked in config -- but no obvious location (.cmus, /config/cmus, .config/cmus/rc... nada).

I'll add a new ticket on building issues, to spread the request.

Meanwhile, it *definitely* is an issue if cmus can't follow a symlink. If that's the root cause, it's a dealbreaker! But I bet there is something else going on.....

**Comment by thewaywest** [comment]
ohhh, look at that! the files identified as <5s length all have some sort of problem with accented characters. The album of blank Erik Satie files I had is listed under "Erik Satie; Pascal Rog<e9>", and is _parallel to_ a correclt registered album "Erik Satie; Pascal Rogé" Like wise, other non-FLAC files are variously duplicated or problematic.

This is a problem with diacritics!

This should be handled by default... but maybe there is a setting that manages it. Thoughts?

**Comment by thewaywest** [comment]
dang. 

- I cleared my install and deleted my local config file, to start from scratch. 
- I built from source (probably didn't have to, though)
- Instead of adding from my home directory Music folder (which is itself a symlink to another partition), I added directly to the mounted partition: /media/home/Music
- problem solved (sort of)

... as mahkoh was trying to tell me at the start of the day. So the immediate issue is resolved, but the deeper question is "why"?

since we are running in bash, i wonder if the is a syntax problem when accessing files. Here's an interesting lead from a couple of years ago...

https://unix.stackexchange.com/questions/102464/cant-echo-or-create-a-symbolic-link-to-a-file-with-special-characters-in-its-na
